src/openrct2/ride/Vehicle.cpp

```diff
@@ -5506,14 +5505,18 @@ void Vehicle::UpdateCrash()
                     ExplosionCloud::Create({ curVehicle->x + xOffset, curVehicle->y + yOffset, curVehicle->z });
                 }
             }
-            if (curVehicle->animationState + 7281 > 0xFFFF)
+            if (curVehicle->animationState <= 58248)
```

Comment:
Can you switch to using the hex values for this as it will be easier to understand. I'm not sure this logic is still the same due to the size of scenario_rand